### PR TITLE
No Errors Button

### PR DESCRIPTION
**Pull request by Adam-Roney581**
### What this PR does :thinking: 
Adds a no errors button to let the user know the sortie is error-free.

### Resolves :white_check_mark: 
[PBS-3375](https://jira.il2.dso.mil/browse/PBS-3375)

### Testing Items :microscope: 
- [ ] Add some errors to a sortie and then fixed them
- [ ] Green checkmark button should appear where the error/warning button was

### Notes :writing_hand: 
I had to check the box for several files to add their "Target Membership" to the UI and Unit test projects. Not sure if this a new xcode thing because I don't know how they were working before with that being checked. 

### UI Changes :eye: 
New checkmark button.

### CODE REVIEW

File: PBLogging/ViewModels/PlaceholderContentViewModel.swift [hunk 1]
```diff
@@ -12,4 +12,5 @@ struct PlaceholderContentViewModel {
     var image: Image
     var title: String
     var details: String?
+    var imageColor: Color?
```
Comment (tsanders88) [inline PBLogging/ViewModels/PlaceholderContentViewModel.swift:15]: not used?

File: PBLogging/Views/RecentSorties/RecentSortiesList.swift [hunk 2]
```diff
@@ -26,6 +26,7 @@ struct RecentSortiesList: View {
     @State var flagsRefresh: [Bool]
     @State var isExpanded: [Bool]
     @State var showFlagsButton = false
+    @State var showNoErrorsPopover = false
```
Comment (tsanders88) [inline PBLogging/Views/RecentSorties/RecentSortiesList.swift:29]: not used?

File: PBLogging/Views/SummaryCards/PlaceholderSummaryContentView.swift [hunk 3]
```diff
@@ -14,7 +14,7 @@ struct PlaceholderSummaryContentView: View {
         VStack(spacing: 8) {
             placeholder.image
                 .font(.title2)
-                .foregroundColor(.primary.opacity(0.6))
+                .foregroundColor(placeholder.imageColor != nil ? placeholder.imageColor : .primary.opacity(0.6))
```
Comment (tsanders88) [inline PBLogging/Views/SummaryCards/PlaceholderSummaryContentView.swift:17]: can you do placeholder.imageColor ?? .primary.opacity(0.6) or is that a .NET thing

File: PBLogging/Components/SortieFlags/NoErrorsButton.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
+//
+//  NoErrorsButton.swift
+//  PBLogging
+//
+//  Created by Adam Roney on 1/6/23.
+//
+
+import SwiftUI
+
+struct NoErrorsButton: View {
```
Comment (tsanders88) [inline PBLogging/Components/SortieFlags/NoErrorsButton.swift:10]: might want to just incorporate into sortie flags button

File: PBLogging/ViewModels/PlaceholderContentViewModel.swift (hunk 1 above)
Comment (Adam-Roney581) [reply PBLogging/ViewModels/PlaceholderContentViewModel.swift:15]: It's used in the PlaceholderSummaryContentView passed from the NoErrorsButton

File: PBLogging/Views/RecentSorties/RecentSortiesList.swift (hunk 2 above)
Comment (Adam-Roney581) [reply PBLogging/Views/RecentSorties/RecentSortiesList.swift:29]: good catch, removed

File: PBLogging/Views/SummaryCards/PlaceholderSummaryContentView.swift (hunk 3 above)
Comment (Adam-Roney581) [reply PBLogging/Views/SummaryCards/PlaceholderSummaryContentView.swift]: You can. I changed it. Should've known about that. Good catch

File: PBLogging/Components/SortieFlags/NoErrorsButton.swift (hunk 4 above)
Comment (Adam-Roney581) [reply PBLogging/Components/SortieFlags/NoErrorsButton.swift]: Good idea. I re-factored it. Can you test again?

**Review by tsanders88 — APPROVED** [review]

**Pull request merged**